custom/src/main/java/com/solarwinds/opentelemetry/extensions/initialize/config/HttpSettingsReaderDelegate.java

```diff
@@ -0,0 +1,98 @@
+package com.solarwinds.opentelemetry.extensions.initialize.config;
+
+import com.solarwinds.joboe.logging.Logger;
+import com.solarwinds.joboe.logging.LoggerFactory;
+import com.solarwinds.joboe.sampling.Settings;
+import com.solarwinds.joboe.shaded.google.gson.Gson;
+import com.solarwinds.joboe.shaded.google.gson.GsonBuilder;
+import com.solarwinds.opentelemetry.extensions.config.JsonSetting;
+import com.solarwinds.opentelemetry.extensions.config.JsonSettingWrapper;
+import io.opentelemetry.api.internal.InstrumentationUtil;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class HttpSettingsReaderDelegate {
+
+  private static final Logger logger = LoggerFactory.getLogger();
+
+  private static final Gson gson = new GsonBuilder().create();
+
+  public Settings fetchSettings(String url, String authorizationHeader) {
+    AtomicReference<Settings> settings = new AtomicReference<>();
+    InstrumentationUtil.suppressInstrumentation(
+        () -> {
+          HttpURLConnection connection = null;
+          try {
+            connection = getHttpURLConnection(new URL(url), authorizationHeader);
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+              String errorResponse = getErrorMessage(connection);
+              logger.error(
+                  String.format(
+                      "HTTP request failed with status code: %s and error: %s",
+                      responseCode, errorResponse));
+            } else {
+              String payload;
+              try (BufferedReader reader =
+                  new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+                StringBuilder responseBuilder = new StringBuilder();
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                  responseBuilder.append(line);
+                }
+                payload = responseBuilder.toString();
+              }
+
+              settings.set(JsonSettingWrapper.wrap(gson.fromJson(payload, JsonSetting.class)));
```

Comment:
Couldn't the `BufferedReader` just be passed to `fromJson` here ?

Reply:
it could, thanks!